ngraph/CMakeLists.txt

```diff
@@ -124,6 +124,7 @@ option(NGRAPH_ADDRESS_SANITIZER_ENABLE "Compiles and links with Address Sanitize
 option(NGRAPH_THREAD_SANITIZER_ENABLE "Compiles and links with Thread Sanitizer" FALSE)
 option(NGRAPH_UB_SANITIZER_ENABLE "Compiles and links with Undefined Behavior Sanitizer" FALSE)
 option(NGRAPH_USE_PROTOBUF_LITE "Compiles and links with protobuf-lite" FALSE)
+option(NGRAPH_STATIC_LIBRARY "Compiles ngraph as a static library" FALSE)
```

Comment:
I think we don't need a separate `NGRAPH_STATIC_LIBRARY` since we can use standard `BUILD_SHARED_LIBS`

Reply:
I wanted to leave the flexibility of having a standalone inference_engine.dll (with ngraph static) or go with full static. But as you wish :)

Reply:
`BUILD_SHARED_LIBS` with your change allows to have only ngraph to be `STATIC` / `DYNAMIC`, while `inference_engine` has hardcoded `SHARED` library type

Reply:
Sure but if in a next PR i propose a change to allow inference_engine and its plugins to be in a static library, do you think there is a use case where we don't want everything to be static ?

Reply:
I think we need only all static or all dynamic. We don't need to play with different variations where something is static, others are dynamic.